@here/harp-mapview/lib/MapView.ts

```diff
@@ -1151,6 +1152,11 @@ export class MapView extends EventDispatcher {
 
         this.m_themeManager = new MapViewThemeManager(this, this.m_uriResolver);
 
+        this.m_poiRenderer = new PoiRenderer(this.m_renderer, this.m_poiManager, [
+            this.imageCache,
+            this.userImageCache
+        ]);
+
```

Comment:
Just some notes for further refactoring on a separate PR, i guess this one is already too large:

MapView shouldn't now about PoiRenderer, actually it shouldn't even know about TextElementsRenderer. We should move to an MapView abstraction similar to that in HARP Native, where you only have data sources and buckets, and some of them may handle text.

If we want to support smooth transitions between themes (at least Roberto is working on that), we'll probable have too keep the TextElementsRenderer alive as well.

Reply:
Good point... we should definitly go on clearing things up here, on the long run it will save us time and nerves if we do, but as you said, this refactor is getting already very big, and there is so much more that could be done.